metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -211,6 +226,7 @@ class BuildServerConnection private (
       register(
         server => server.buildTargetScalaMainClasses(params),
         onFail,
+        defaultTimeout("main classes"),
```

Comment:
One thing I realised is that, if a user is hit with that message, they wouldn't know what is going on. Most likely it's enough to show notification on compilation and otherwise cancel main/test automatically with a warning. What do you think?

Reply:
Agreed. Let's not ask in such case.